packages/core/js-client/src/index.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ClientConfig, IFluenceClient, RelayOptions, ConnectionState, CallAquaFunctionType, RegisterServiceType } from '@fluencelabs/interfaces';
+import { ClientPeer, makeClientPeerConfig } from './clientPeer/ClientPeer.js';
+import { callAquaFunction } from './compilerSupport/callFunction.js';
+import { registerService } from './compilerSupport/registerService.js';
+import { MarineBackgroundRunner } from './marine/worker/index.js';
+// @ts-ignore
+import { BlobWorker, Worker } from 'threads';
+import { doRegisterNodeUtils } from './services/NodeUtils.js';
+import { fetchResource } from './fetchers/index.js';
+import process from 'process';
+
+import avmWasmUrl from '../node_modules/@fluencelabs/avm/dist/avm.wasm?url';
+import marineJsWasmUrl from '../node_modules/@fluencelabs/marine-js/dist/marine-js.wasm?url';
+import workerCodeUrl from '../node_modules/@fluencelabs/marine-worker/dist/__ENV__/marine-worker.umd.cjs?url';
+
+const JS_CLIENT_VERSION = '__JS_CLIENT_VERSION__';
+
+const isNode = typeof process !== 'undefined' && process?.release?.name === 'node';
```

Comment:
would it be useful to have this as a function that can be reused not repeat yourself? I even rememeber a lib was used somewhere for that, but a lib can be an overkill here

if it's hard to share this code - leave it as it is, not critical at all

Reply:
I will remove this function a bit later.
The good place for this kind of function is 'fetcher' folder.